### ISSUE TITLE
[Discussion] What does `let` do?

### ISSUE THREAD

**Issue by Dorus** [Reactive-Extensions/RxJS#1156]
Mostly by my own mistake, I've been using `let(selector)` as an alias of `publish(selector)` where it clearly isn't.

Now I'm wondering what the following sentence means:

> Selector function which can use the source sequence as many times as needed, **without sharing subscriptions to the source sequence.**

It say subscriptions are not shared to the source.  Also, on [reactivex.io](http://reactivex.io/documentation/operators/publish.html), `let` is listed under `publish`. But the when you read closely, it say "does not multicast".

And indeed `let` does not multicast. So probably "without sharing subscriptions to the source sequence" means it does not share a single subscription to the source, but multiple.

I think 2 things can be made more clear in the docs. Namely that the `let` operator does not multicast, and that `publish(func)` is an alternative if you do need multicast. And also (I hope i guessed right) that the `func` parameter in `let` should still return an subscriptions that unsubscribe all inner observables.

**Comment by trxcllnt** [comment]
@Dorus you're right, the listing for `let` erroneously points to the page for `publish`, thanks for pointing that out.

I think it'll clear things up to take a look at the [implementation](https://github.com/Reactive-Extensions/RxJS/blob/master/src/core/linq/observable/let.js) of the `let` operator. `let` immediately calls the selector function with `this` and returns the result, which means examples [1] and [2] below are identical:

```es6
function createAddOneObservable (source) {
  return source.map((x) => x + 1);
}

const interval = Observable.range(10);

// 1.
interval.let(createAddOneObservable).subscribe();

// 2.
createAddOneObservable(interval).subscribe();
```
